kubetest2/kubetest2-gce/deployer/up.go

```diff
@@ -135,3 +93,75 @@ func enableComputeAPI(project string) error {
 
 	return nil
 }
+
+// getProjectFromBoskos creates a boskos client, acquires a gcp project
+// and starts a heartbeat goroutine to keep the project reserved
+func (d *deployer) getProjectFromBoskos() error {
+	boskos, err := client.NewClient(
+		os.Getenv("JOB_NAME")+"-kubetest2",
+		d.BoskosLocation,
+		"",
+		"",
+	)
+	if err != nil {
+		return fmt.Errorf("failed to create boskos client: %s", err)
+	}
+
+	resourceType := "gce-project"
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute*5)
+	defer cancel()
+
+	boskosProject, err := boskos.AcquireWait(ctx, resourceType, "free", "busy")
+	if err != nil {
+		return fmt.Errorf("failed to get a %q from boskos: %s", resourceType, err)
+	}
+	if boskosProject == nil {
+		return fmt.Errorf("boskos had no %s available", resourceType)
+	}
+
+	startBoskosHeartbeat(
+		boskos,
+		boskosProject,
+		time.Duration(d.BoskosAcquireTimeoutSeconds)*time.Second,
+		d.boskosHeartbeatClose,
+	)
+
+	d.boskos = boskos
+	d.boskosProject = boskosProject
+	d.GCPProject = boskosProject.Name
+
+	return nil
+}
+
+// startBoskosHeartbeat starts a goroutine that sends periodic updates to boskos
+// about the provided resource until the channel is closed. This prevents
+// boskos from taking the resource from the deployer while it is still in use.
```

Comment:
reaper from taking the resource ... 
to be specific

also we only care about heartbeat in the CI use case, i don't think we should really care about users running reaper/janitor locally.

Reply:
Updated the comment. I'd like to avoid a split between CI and non-CI behavior for as long as I can, and this seems like a pretty harmless goroutine to leave running.